cmake/SetupBlaze.cmake

```diff
@@ -19,24 +19,32 @@ set_property(TARGET Blaze PROPERTY
   INTERFACE_INCLUDE_DIRECTORIES ${BLAZE_INCLUDE_DIR})
 set_property(TARGET Blaze PROPERTY
   INTERFACE_LINK_LIBRARIES Lapack)
+target_link_libraries(
+  Blaze
+  INTERFACE
+  Blas
+  GSL::gsl # for BLAS header
+  Lapack
+  )
 
 target_compile_definitions(Blaze
   INTERFACE
-  # Override SMP configurations
+  # Enable external BLAS kernels
+  BLAZE_BLAS_MODE=1
+  # Set default matrix storage order to column-major
+  BLAZE_DEFAULT_STORAGE_ORDER=blaze::columnMajor
+  # Disable SMP parallelization
   BLAZE_USE_SHARED_MEMORY_PARALLELIZATION=0
-  BLAZE_OPENMP_PARALLEL_MODE=0
-  BLAZE_CPP_THREADS_PARALLEL_MODE=0
-  BLAZE_BOOST_THREADS_PARALLEL_MODE=0
   # Disable MPI parallelization
   BLAZE_MPI_PARALLEL_MODE=0
-  # Disable HPX parallelization
-  BLAZE_HPX_PARALLEL_MODE=0
-  BLAZE_USE_PADDING=0
-  # Enable non-temporal stores for cache optimization of large data structures
-  BLAZE_USE_STREAMING=1
```

Comment:
please keep these optimizations enabled

Reply:
These are enabled by default. I kept `BLAZE_USE_STREAMING` explicitly enabled for you. I think it's not really necessary, but I don't feel strongly about it. I split this change into a separate commit.